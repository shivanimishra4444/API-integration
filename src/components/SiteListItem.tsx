import * as React from 'react'
import { Link } from 'react-router-dom'
import ISite from 'types'

interface IProps {
  site: ISite
}
export default class SiteListItem extends React.Component<IProps, any> {
  public render() {
    return (
      <li className="list-group-item d-flex justify-content-between align-items-center">
        {this.props.site.name}

        <Link className="btn btn-success" to={{ pathname: `/detail/${this.props.site.id}`, state: { site: this.props.site } }}>
          Detail
        </Link>
      </li>
    )
  }
}
