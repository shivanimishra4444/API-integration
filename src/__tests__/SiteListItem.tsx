import * as React from 'react'
import SiteListItem from '../components/SiteListItem'
import * as Adapter from 'enzyme-adapter-react-16'
import { shallow, configure } from 'enzyme'
configure({ adapter: new Adapter() })

const ISite = {
  website: 'abc',
  language_code: 'string',
  currency: 'string',
  name: 'string',
  platform: 'string'
}
describe('<SiteListItem />', () => {
  const wrapper = shallow(<SiteListItem key="123" site={ISite} />)

  it('matches snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })
  it('render 1 <SiteList> component snapshot', () => {
    expect(wrapper).toHaveLength(1)
  })
})
