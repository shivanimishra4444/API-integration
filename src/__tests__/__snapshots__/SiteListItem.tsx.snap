// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SiteListItem /> matches snapshot 1`] = `ShallowWrapper {}`;
